src/Valit/Rules/ValitRules.cs

```diff
@@ -66,20 +67,26 @@ IValitResult IValitRules<TObject>.Validate(Func<IValitRule<TObject>, bool> predi
 
         private IValitResult Validate(IEnumerable<IValitRule<TObject>> rules)
         {
-            var result = ValitResult.CreateSucceeded();
+            var result = ValitResult.Success;
 
             foreach(var rule in rules.ToList())
             {
                 result &= rule.Validate(_object);
 
-                if(_strategy == ValitRulesStrategies.FailFast && !result.Succeeded)
+                if(!result.Succeeded)
                 {
-                    break;
+                    bool cancel = false;
+                    _strategy.Fail(rule, result, ref cancel);
```

Comment:
can't we replace it with out?
```
_strategy.Fail(rule, result, out bool cancel);
if(cancel)
    break;
```
and in 
```
Fail( _, _, out bool cancel)
{
    ...
    cancel = true | false;
}
```

Reply:
I am fine with both. @GooRiOn , please decide. :)

I only didn't want to return bool to signify cancel as it might riddle some people.
4th - acceptable - option would be to return some strict enum or struct.

Reply:
I'd go for **out** as well 👍